active/ScrollingView/ScrollingView.md

```diff
@@ -0,0 +1,2584 @@
+}
+```
+
+### Jump by some delta of the current offset (no animation)
+
+A custom horizontal list control with a `ScrollingView` in its control template jumps the horizontal offset by special values when processing the Left and Right Arrow keystrokes, for a FlipView-like experience.
+
+The ScrollBy method call has no visual effect when attempting to jump beyond the list control's extent.
+
+*`ScrollingView` (new)*
+
+```csharp
+protected override void OnKeyUp(KeyRoutedEventArgs e)
+{
+    base.OnKeyUp(e);
+    if (e.Key == VirtualKey.Left)
+    {
+        _scrollingView.ScrollBy(
+            horizontalOffset: -_previousGroup.Width,
+            verticalOffset: 0.0,
+            new ScrollingScrollOptions(ScrollingAnimationMode.Disabled));
+    }
+    else if (e.Key == VirtualKey.Right)
+    {
+        _scrollingView.ScrollBy(
+            horizontalOffset: _currentGroup.Width,
+            verticalOffset: 0.0,
+            new ScrollingScrollOptions(ScrollingAnimationMode.Disabled));
+    }
+}
+```
+
+*ScrollViewer (existing)*
+
+```csharp
+protected override void OnKeyUp(KeyRoutedEventArgs e)
+{
+    base.OnKeyUp(e);
+    if (e.Key == VirtualKey.Left)
+    {
+        _scrollViewer.ChangeView(
+            horizontalOffset: _scrollViewer.HorizontalOffset - _previousGroup.Width),
+            verticalOffset: null,
+            zoomFactor: null,
+            disableAnimation: true);
+    }
+    else if (e.Key == VirtualKey.Right)
+    {
+        _scrollViewer.ChangeView(
+            horizontalOffset: _scrollViewer.HorizontalOffset + _currentGroup.Width),
+            verticalOffset: null,
+            zoomFactor: null,
+            disableAnimation: true);
+    }
+}
+```
+
+### Animating with a custom curve to a new offset by some delta of the current
+
+A Slider's Value changes animate the `ScrollingView`'s VerticalOffset with custom animation durations.
+
+*`ScrollingView` (new)*
+
+```csharp
+// This variable is used to keep track of the ScrollBy operations triggered by interactions with the Slider.
+// Other programmatic view changes, which also trigger the ScrollAnimationStarting event, will not use a 
+// custom curve because their ScrollInfo are not recorded in the hash table.
+private HashSet<ScrollingScrollInfo> _scrollsWithCustomDuration = new HashSet<ScrollingScrollInfo>();
+
+private void VerticalOffsetSlider_ValueChanged(object sender, RangeBaseValueChangedEventArgs e)
+{
+    double verticalOffsetDelta = GetOffsetDelta();
+    ScrollingScrollInfo scrollInfo = _scrollingView.ScrollBy(0.0, verticalOffsetDelta);
+    _scrollsWithCustomDuration.Add(scrollInfo);
+}
+
+private void ScrollingView_ScrollAnimationStarting(ScrollingView scrollingView, ScrollingScrollAnimationStartingEventArgs e)
+{
+    if (_scrollsWithCustomDuration.Contains(e.ScrollInfo))
+    {
+        TimeSpan animationDuration = GetAnimationCustomDuration(e.StartPosition, e.EndPosition);
+        (e.Animation as Vector3KeyFrameAnimation).Duration = animationDuration;
```

Comment:
Discussion: difficult to use because you don't know what type the animation is. Note that you can overwrite the animation, but you have to know how to get it set up (the one that's in there now is pre-configured somewhat)

Add a type check to validate the animation type? 

Can we make it a KeyFrameAnimation? That might be too limiting.

Document that this is always a KeyFrameAnimation when the event is first raised?

Should we add a method to the args to let you set the duration without having to know what the animation is?

What is the typical/common expected use case for this?

Recommend: Document that the Animation property always starts as a Vector3KeyFrameAnimation, but explain that you can overwrite it with a new animation.